include/xtensor/xtensor.hpp

```diff
@@ -548,6 +549,96 @@ namespace xt
 
         return out;
     };
+
+    /**
+     * Converts ``std::vector<index_type>`` (returned e.g. from ``xt::argwhere``) to a flattened
+     * ``xtensor``.
+     *
+     * @param vector of indices
+     *
+     * @return ``xt::xtensor<typename index_type::value_type, 1>`` (e.g. ``xt::xtensor<size_t, 1>``)
+     */
+    template <class T>
+    inline auto flatten_indices(const std::vector<T> &idx)
+    {
+        auto n = idx.size();
+        if (n != 0)
+        {
+            n *= idx[0].size();
+        }
+
+        using return_type = xtensor<typename T::value_type, 1>;
+        return_type out = xtensor<typename T::value_type, 1>::from_shape({n});
```

Comment:
Same remark here (either use `return_type` or remove it since it's use only once)